honeybee_radiance/command/options/oconv.py

```diff
@@ -0,0 +1,115 @@
+"""Oconv parameters."""
+from .optionbase import OptionCollection, FileOption, TupleOption, IntegerOption, \
+    BoolOption
+
+
+class OconvOptions(OptionCollection):
+    """
+    [ -i octree | -b xmin ymin zmin size ][ -n objlim ][ -r maxres ][ -f ][ -w ]
+
+    Also see: https://floyd.lbl.gov/radiance/man_html/oconv.1.html
+    """
+
+    __slots__ = ('_i', '_b', '_n', '_r', '_f', '_w')
+
+    def __init__(self):
+        """oconv command options."""
+        OptionCollection.__init__(self)
+        self._i = FileOption('i', 'existing octree file')
+        self._b = TupleOption(
+            'b', 'bounding box as min_x, min_y, min_z and size', None, 4, float
+        )
+        self._n = IntegerOption(
+            'n', 'maximum surface set size for each voxel - default: 6', min_value=0
+        )
+        self._r = IntegerOption(
+            'r', 'maximum octree resolution - default: 16384', min_value=0
+        )
+        self._f = BoolOption('f', 'frozen octree - default: off')
+        self._w = BoolOption('w', 'warning messages - default: on')
+        self._on_setattr_check = True
+
+    def on_setattr(self):
+        assert not (self.b.is_set and self.i.is_set), \
+            'The -b and -i options are mutually exclusive.'
+
+    @property
+    def i(self):
+        """input octree."""
+        return self._i
+
+    @i.setter
```

Comment:
If you aren't performing any checks on the setters of this class do you actually need them as setters?  Can't they just be attributes?  I am also a bit unclear on what type of object they are supposed to be. Is is a string?

Reply:
The check is done in line 18. We still need a setter to ensure the check happens.